### ISSUE TITLE
hangs on boot when GPG key missing

### ISSUE THREAD

**Issue by ncfavier** [Mic92/sops-nix#521]
If the GPG private key used for decrypting secrets is missing, the boot just hangs after "setting up secrets". Is this intentional?

**Comment by Mic92** [comment]
No. This might be caused by the recent switch to gnupg instead of using it's own gpg implementation. What is your set up?

**Comment by ncfavier** [comment]
I experienced this with sops-nix 73bf36912e31a6b21af6e0f39218e067283c67ef, nixpkgs 97b17f32362e475016f942bbdfda4a4a72a8a652. Note that I'm overriding `SOPS_GPG_EXEC` [here](https://github.com/ncfavier/config/blob/9b0f7aacd6615be24e6646e569ede8ecaf0620e7/modules/secrets.nix#L16-L19) for reasons I don't fully remember.
